src/transformers/models/bart/modeling_bart.py

```diff
@@ -407,7 +407,7 @@ def forward(
         hidden_states: torch.Tensor,
         encoder_hidden_states: torch.Tensor,
         encoder_attn_mask: Optional[torch.Tensor] = None,
-        past_key_value: Optional[Tuple[Tuple[torch.Tensor]]] = None,
+        past_key_value: Optional[Tuple[torch.Tensor]] = None,
```

Comment:
the `past_key_value` should have exactly one level for each layer, no matter whether the model is a decoder-only a.k.a. GPT2 or BART.

For GPT2, the tuple for each layer contains 2 tensors: key and value states
For BART/T5, the tuple for each layer contains 4 tensors: key and value states of uni-directional self-attention, saved key and value states for cross-attention

Reply:
cc @patil-suraj for information